tests/generators/TestVectorGenerator.ts

```diff
@@ -0,0 +1,55 @@
+import Jwk from "../../lib/core/versions/latest/util/Jwk";
+import Multihash from "../../lib/core/versions/latest/Multihash";
+import OperationGenerator from "./OperationGenerator";
+
+export default class TestVectorGenerator {
+    /**
+     * Generate and prints out test vectors for operation requests
+     */
+    public static async generateOperationVectors() {
+
+        // generate a create operation request
+        const createOperationData = await OperationGenerator.generateCreateOperation();
+
+        // derive an update operation request from the create
+        const [nextUpdateKey] = await OperationGenerator.generateKeyPair('nextUpdateKey');
+        const nextUpdateCommitmentHash = Multihash.canonicalizeThenDoubleHashThenEncode(nextUpdateKey.jwk);
+        const [anyNewSigningKey] = await OperationGenerator.generateKeyPair('newKeyId');
+        const patches = [
+          {
+            action: 'add-public-keys',
+            public_keys: [
+              anyNewSigningKey
+            ]
+          }
+        ];
+        const updateRequest = await OperationGenerator.createUpdateOperationRequest(
+            createOperationData.createOperation.didUniqueSuffix,
+            createOperationData.updatePublicKey,
+            createOperationData.updatePrivateKey,
+            nextUpdateCommitmentHash,
+            patches
+        )
+        
+        // derive a recover operation from the create operation
+        const [newRecoveryPublicKey] = await Jwk.generateEs256kKeyPair();
+        const [newSigningPublicKey] = await OperationGenerator.generateKeyPair('keyAfterRecover');
+
+        const [documentKey] = await OperationGenerator.generateKeyPair('newDocumentKey');
+        const newServiceEndpoints = OperationGenerator.generateServiceEndpoints(['newId']);
+  
+        const recoverOperationRequest = await OperationGenerator.generateRecoverOperationRequest(
```

Comment:
Sanity: what about deactivate operation?

Reply:
Lol good point. This change didn't touch deactivate so I forgot about it. Will add